test/e2e/canary_test.go

```diff
@@ -112,6 +112,58 @@ func (s *CanarySuite) TestRolloutScalingWhenPaused() {
 		ExpectCanaryStablePodCount(1, 3)
 }
 
+
+// TestRolloutWithMaxSurgeScalingDuringUpdate verifies behavior when scaling a rollout up/down in middle of update and with maxSurge 100%
+func (s *CanarySuite) TestRolloutWithMaxSurgeScalingDuringUpdate() {
+	s.Given().
+		HealthyRollout(`
+apiVersion: argoproj.io/v1alpha1
+kind: Rollout
+metadata:
+  name: updatescaling
+spec:
+  replicas: 4
+  strategy:
+    canary:
+      maxSurge: 100
```

Comment:
I think you mean for this to be "100%" and not `100`. The latter is an absolute number.

Reply:
yes..meant 100%